dashboard/app/controllers/api_controller.rb

```diff
@@ -442,7 +442,27 @@ def course_summary
 
   def unit_summary
     unit_name = params[:unit_name]
-    unit = Unit.get_from_cache(unit_name)
+    unit_position = params[:unit_position]
+    course_name = params[:course_name]
+    unit = nil
+
+    # unit_summary supports calls with either :unit_name or :course_name AND :unit_position
+    if unit_name
+      context = Queries::Courses.get_course_context(unit_name)
+      if context
+        unit = context[:unit]
+        course_name = context[:course].name
+        unit_position = context[:unit_group_unit].position
+      else
+        # Unit hasn't been migrated to have a Course/UnitGroup
```

Comment:
Are we tracking removing this code after everything has been migrated? What's the timeline on the migration?

Reply:
Good idea to track this work: https://codedotorg.atlassian.net/browse/TEACH-1755.

The migration will probably be in a month or two after we have the main feature work done and everything is stable. Right now our highest priority is getting modularity ready for curriculum writers so they aren't blocked.